x-pack/plugins/reporting/server/lib/store/store.ts

```diff
@@ -71,6 +71,14 @@ const jobDebugMessage = (report: Report) =>
   `[attempts: ${report.attempts}] ` +
   `[process_expiration: ${report.process_expiration}]`;
 
+const getEventLogger = (reporting: ReportingCore, report: Report) => {
```

Comment:
I'm thinking maybe we don't need this function. Maybe I'm wrong, but I'm thinking if:

1. If we didn't have this `getEventLogger` method, instead the code would use `reporting.getEventLogger(report)`.
2. Then inside the `ReportingEventLogger` we would not need to store `eventObj`, instead we would just store the reference to `report`.
3. And then in every logging method (`logRetry`, `logReportSaved`, etc..) we would not need to use `deepMerge`, but we could construct the necessary object directly, from the `report` reference.

Reply:
https://github.com/elastic/kibana/pull/119923/commits/c17b35d3acdbed2c82272edaa2a0f9f601cc1fe1